include/boost/geometry/strategies/cartesian/side_rounded_input.hpp

```diff
@@ -0,0 +1,70 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2021 Tinko Bartels, Berlin, Germany.
+
+// Use, modification and distribution is subject to the Boost Software License,
+// Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+
+#ifndef BOOST_GEOMETRY_STRATEGY_CARTESIAN_SIDE_ROUNDED_INPUT_HPP
+#define BOOST_GEOMETRY_STRATEGY_CARTESIAN_SIDE_ROUNDED_INPUT_HPP
+
+#include <boost/geometry/core/config.hpp>
+
+#include <boost/geometry/strategies/side.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+namespace strategy { namespace side
+{
+
+
+template <typename CalculationType = void, int Coeff1 = 5, int Coeff2 = 32>
+struct side_rounded_input
+{
+    using cs_tag = cartesian_tag;
+
+    template <typename P1, typename P2, typename P>
+    static inline int apply(P1 const& p1, P2 const& p2, P const& p)
+    {
+        using coordinate_type1 = typename coordinate_type<P1>::type;
+        using coordinate_type2 = typename coordinate_type<P2>::type;
+        using coordinate_type3 = typename coordinate_type<P>::type;
+
+        using coor_t = std::conditional_t
+            <
+                std::is_void<CalculationType>::value,
+                typename select_most_precise
+                    <
+                        coordinate_type1,
+                        coordinate_type2,
+                        coordinate_type3
+                    >::type,
+                CalculationType
+            >;
```

Comment:
All of these lines could be replaced with:
```
using coor_t = typename select_calculation_type_alt<CalculationType, P1, P2, P>::type;
```
from here:
https://github.com/boostorg/geometry/blob/develop/include/boost/geometry/util/select_calculation_type.hpp#L54-L70